features/storage/TESTS/blockdevice/heap_block_device/main.cpp

```diff
@@ -23,6 +23,11 @@
 
 using namespace utest::v1;
 
+// TODO HACK, replace with available ram/heap property
+#if defined(TARGET_LPC1114)
```

Comment:
This does not seem right. Replace `TARGET_LPC1114` by what is preventing the target to run the test.
Do the same everywhere you have prevented a test from running.

Reply:
@hugueskamba Thanks for your comment.  How do you think about this?

```
#define TEST_BLOCK_SIZE 128
#define TEST_BLOCK_DEVICE_SIZE 32*TEST_BLOCK_SIZE
#define TEST_BLOCK_COUNT 10
#define TEST_ERROR_MASK 16

#if (MBED_RAM_SIZE <= TEST_BLOCK_DEVICE_SIZE)
#error [NOT_SUPPORTED] Insufficient heap for heap block device tests
#else
```

Reply:
I am not sure why you this test needs to be excluded for that board.
Here is a question, why do you need to exclude this test for that board?
If the reason is that the size of the block device is greater than the available RAM then it is fine.

Reply:
Because the LPC1114 RAM size is 4kB which is not suitable to run this test case.

```
#define TEST_BLOCK_SIZE 128
#define TEST_BLOCK_DEVICE_SIZE 32*TEST_BLOCK_SIZE

(snip)

void test_read_write()
{
    uint8_t *dummy = new (std::nothrow) uint8_t[TEST_BLOCK_DEVICE_SIZE];
```

Reply:
> Because the LPC1114 RAM size is 4kB which is not suitable to run this test case.

Then have that in your pre-processor macro.